data/Store.js

```diff
@@ -786,21 +798,56 @@ export class Store extends HoistBase {
         return recordMap;
     }
 
-    parseFieldValues(data, skipMissingFields = false) {
-        const ret = {};
+    parseRaw(data) {
+        const {shareDefaults} = this.experimental,
+            ret = shareDefaults ? Object.create(this._dataDefaults) : {};
         this.fields.forEach(field => {
-            const {name} = field;
-
-            // Sometimes we want to ignore fields which are not present in the data to preserve
-            // an undefined value, to allow merging of data with existing data. In these cases we
-            // do not want the configured default value for the field to be used, as we are dealing
-            // with a partial data object
-            if (skipMissingFields && !has(data, field.name)) return;
+            const {name} = field,
+                val = field.parseVal(data[name]);
+            if (!shareDefaults || val !== field.defaultValue) {
+                ret[name] = val;
+            }
+        });
+        return ret;
+    }
 
-            ret[name] = field.parseVal(data[name]);
+    parseUpdate(data, update) {
+        const {shareDefaults} = this.experimental,
+            ret = shareDefaults ? Object.create(this._dataDefaults) : {};
+        this.fields.forEach(field => {
+            const {name} = field,
+                updateVal = update[name];
+            if (!isUndefined(updateVal)) {
+                const val = field.parseVal(updateVal);
+                if (!shareDefaults || val !== field.defaultValue) {
+                    ret[name] = val;
+                }
+            } else {
+                const existingVal = data[name];
+                if (!isUndefined(existingVal)) {
```

Comment:
Is there any value in also checking if `!sharedDefaults`? 

I believe that, in that (default) case we can count on no `undefined` values being present in existing `data` - i.e. `data` will be complete, with a field for all keys and no `undefined` values for those keys. 

That said, it still might not be a bad idea for symmetry and to ensure that `ret` always has the full set of fields when shared defaults not in use.

Reply:
great idea, done